### ISSUE TITLE
Add a new output format option to print errors as GitHub Actions Workflow commands 

### ISSUE THREAD

**Issue by edgarrmondragon** [python/mypy#17612]
**Feature**

Add a new `--output=github` option to format errors as Workflow commands.

**Pitch**

For example, for this output in JSON format

```console
$ mypy src --output=json
{"file": "src/mylib/client.py", "line": 759, "column": 20, "message": "Argument 1 to \"export_responses\" of \"Client\" has incompatible type \"str\"; expected \"int\"", "hint": null, "code": "arg-type", "severity": "error"}
```

we'd get the following output in workflow command format

```console
$ mypy src --output=json
::error file=src/mylib/client.py,line=759,col=20::(`arg-type`) Argument 1 to "export_responses" of "Client" has incompatible type "str"
```

**Comment by wyardley** [comment]
This would be super handy

**Comment by webknjaz** [comment]
@edgarrmondragon do you mean using the annotations feature?

**Comment by wyardley** [comment]
I'm not the person who opened the issue, but yes, I think this would support allowing annotations, similar to
`ruff check --output-format github` or using pylint with the `pylint-actions` plugin.

**Comment by edgarrmondragon** [comment]
> @edgarrmondragon do you mean using the annotations feature?

Yeah, these: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions

> I'm not the person who opened the issue, but yes, I think this would support allowing annotations, similar to
> `ruff check --output-format github` or using pylint with the `pylint-actions` plugin.

Exactly!

**Comment by edgarrmondragon** [comment]
A PR is up if anyone's interested: https://github.com/python/mypy/pull/17771

**Comment by mschoettle** [comment]
Based on this comment here (https://github.com/python/mypy/pull/11396#issuecomment-2103437776) I am not sure what the likelihood of this getting accepted will be.

(Until this comment I was hoping to have a code quality/gitlab output in mypy directly instead of relying on an extra plugin)

**Comment by blyedev** [comment]
What's the status of this feature? I'd love to use it in my CI
